job/setupDependencies.js

```diff
@@ -609,17 +696,21 @@ function __getDependencyIntegrations(bag, dependency, next) {
       _.each(stringAndArrayData,
         function (value, key) {
           value = ___escapeEnvironmentVariable(value);
-          bag.commonEnvs.push(util.format('%s_INTEGRATION_%s="%s"',
-            sanitizedDependencyName,
-            key.replace(/[^A-Za-z0-9_]/g, '').toUpperCase(),
-            value
-          ));
+          bag.commonEnvs.push({
+            key: util.format('%s_INTEGRATION_%s',
+              sanitizedDependencyName,
+              key.replace(/[^A-Za-z0-9_]/g, '').toUpperCase()),
+            value: value
+          });
         }
       );
       _.each(objectData,
         function (value, key) {
           value  = ___replaceSingleQuotes(value);
-          bag.commonEnvs.push(util.format('%s=\'%s\'', key, value));
+          bag.commonEnvs.push({
```

Comment:
we should probably add `surroundWithSingleQuotes: true/false` to the object we are inserting to `bag.commonEnvs`. This will help us handle this scenario where the value is surrounded by single quotes.